### PR TITLE
Ports TG's instrument system and guitars

### PR DESCRIPTION
**Pull request by unid15**
Includes cool UI

http://puu.sh/j6m27/c665e92ab9.webm

Guitars can be found in arts&crafts crates
- Other changes: the piano doesn't play itself anymore, and will stop playing music once you go away from it

### CODE REVIEW

**Comment by PJB3005** [comment]
that's not nanoUI.

That's bygax.

Cool nonetheless

**Comment by DeityLink** [comment]
I like it.

File: code/game/objects/structures/musician.dm [hunk 1]
```diff
@@ -245,66 +106,48 @@
...
-		// END AUTOFIX
-		if(song)
-			var/calctempo = (10/song.tempo)*60
-			dat += "Tempo : <A href='?src=\ref[src];tempo=10'>-</A><A href='?src=\ref[src];tempo=1'>-</A> [calctempo] BPM <A href='?src=\ref[src];tempo=-1'>+</A><A href='?src=\ref[src];tempo=-10'>+</A><BR><BR>"
-			var/linecount = 0
-			for(var/line in song.lines)
-				linecount += 1
-				dat += "Line [linecount]: [line] <A href='?src=\ref[src];deleteline=[linecount]'>Delete Line</A> <A href='?src=\ref[src];modifyline=[linecount]'>Modify Line</A><BR>"
-			dat += "<A href='?src=\ref[src];newline=1'>Add Line</A><BR><BR>"
+		dat += "<H3>Editing</H3>"
```
Comment (ririchiyo) [inline code/game/objects/structures/musician.dm]: what is this crap what are you doing

**Comment by ririchiyo** [comment]
You need to convert all these += into {} because strings are slow in BYOND, byjax is also a terrible thing but whatever that's not the point here.

**Comment by PJB3005** [comment]
Yeah it's best if you'd use {} since you're working with in line HTML, you should try to make it as readable as possible, like the Tscience console's bygax version.

File: code/datums/supplypacks.dm [hunk 2]
```diff
@@ -256,8 +256,9 @@ var/list/all_supply_groups = list("Supplies","Clothing","Security","Hospitality"
 	/obj/item/mounted/poster,
 	/obj/item/weapon/wrapping_paper,
 	/obj/item/weapon/wrapping_paper,
-	/obj/item/weapon/wrapping_paper)
-	cost = 10
+	/obj/item/weapon/wrapping_paper,
+	/obj/item/device/instrument/guitar)
```
Comment (wild-billy) [inline code/datums/supplypacks.dm]: stick a tab before the entries in this list while you're at it

File: code/game/objects/items/devices/instruments.dm [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+		user << "<span class='warning'>You don't have the dexterity to do this!</span>"
+		return 1
+	interact(user)
+
+/obj/item/device/instrument/interact(mob/user as mob)
+	if(!user)
+		return
+
+	if(!isliving(user) || user.stat || user.restrained() || user.lying)
+		return
```
Comment (wild-billy) [inline code/game/objects/items/devices/instruments.dm]: there should be a generic proc somewhere to replace the last 4 lines

File: code/game/objects/structures/musician.dm [hunk 4]
```diff
@@ -317,141 +159,176 @@
...
+			playsound(src.loc, 'sound/items/Ratchet.ogg', 50, 1)
+			user << "<span class='notice'> You begin to tighten \the [src] to the floor...</span>"
+			if (do_after(user, 20, target = src))
 				user.visible_message( \
 					"[user] tightens \the [src]'s casters.", \
-					"<span class='notice'>You have tightened \the [src]'s casters. Now it can be played again.</span>", \
-					"You hear ratchet.")
-				src.anchored = 1
+					"<span class='notice'>You tighten \the [src]'s casters. Now it can be played again.</span>", \
+					"<span class='italics'>You hear ratchet.</span>")
```
Comment (wild-billy) [inline code/game/objects/structures/musician.dm]: grammer (--> eg ```You hear a ratchet```)

**Pull request closed without merging**